lib/new_relic/agent/configuration/dotted_hash.rb

```diff
@@ -35,7 +35,7 @@ def self.symbolize(hash)
         def dot_flattened(nested_hash, names=[], result={})
           nested_hash.each do |key, val|
             next if val == nil
-            if val.respond_to?(:has_key?)
+            if val.respond_to?(:has_key?) && !NewRelic::Agent::Configuration::Manager::CONFIG_WITH_HASH_VALUE.include?(key)
```

Comment:
I'm a bit concerned that this could override any cases in the other classes that inherit from `DottedHash` that could return true for this condition.

This seems like a good case to use subclasses to override the functionality of their parent class. This specifies the cases where the `CONFIG_WITH_HASH_VALUE` constant is relevant.

An updated version of `#dot_flattened` could be added to `YamlSource`:
```
        def dot_flattened(nested_hash, names=[], result={})
          nested_hash.each do |key, val|
            next if val == nil
            if val.respond_to?(:has_key?) && 
                NewRelic::Agent::Configuration::Manager::CONFIG_WITH_HASH_VALUE.include?(key)
              add_dot_between_nested_keys(result, names, key, val)
            elsif val.respond_to?(:has_key?)
              dot_flattened(val, names + [key], result)
            else
              add_dot_between_nested_keys(result, names, key, val)
            end
          end
          result
        end
```

The code for the helper method: 
```
# DottedHash (protected section) 

        def add_dot_between_nested_keys(result, names, key, val)
          result[(names + [key]).join('.')] = val
        end
```

Then, the version of `DottedHash#dot_flattened` can remain the same for the other five classes using the method. Thoughts?

Reply:
Yeah, I think that's a great idea. I had totally forgotten that YamlSource and others _inherit_ from this class, not just simply use it. I will plan on making that change! I could move the constant into YamlSource as well, I had a hard time figuring out where it should go. Moving both of those into YamlSource makes the most sense I think